cpp/oneapi/dal/backend/primitives/blas/syrk_dpc.cpp

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/backend/primitives/blas/syrk.hpp"
+#include "oneapi/dal/backend/primitives/blas/misc.hpp"
+
+#include <mkl_dal_sycl.hpp>
+
+namespace oneapi::dal::backend::primitives {
+
+template <typename Float>
+static sycl::event syrk_wrapper(sycl::queue& queue,
+                                mkl::uplo uplo,
+                                mkl::transpose trans,
+                                std::int64_t n,
+                                std::int64_t k,
+                                Float alpha,
+                                const Float* a,
+                                std::int64_t lda,
+                                Float beta,
+                                Float* c,
+                                std::int64_t ldc,
+                                const event_vector& deps) {
+    [[maybe_unused]] const bool is_trans = (trans == mkl::transpose::trans);
+    ONEDAL_ASSERT(ldc >= n);
+    ONEDAL_ASSERT(is_trans || lda >= n);
+    ONEDAL_ASSERT(!is_trans || lda >= k);
+    return mkl::blas::syrk(queue, uplo, trans, n, k, alpha, a, lda, beta, c, ldc, deps);
+}
+
+template <mkl::uplo uplo, typename Float, ndorder ao>
+sycl::event syrk(sycl::queue& queue,
+                 const ndview<Float, 2, ao>& a,
+                 ndview<Float, 2>& c,
+                 Float alpha,
+                 Float beta,
+                 const event_vector& deps) {
+    ONEDAL_ASSERT(a.get_dimension(1) == c.get_dimension(0));
+    ONEDAL_ASSERT(c.get_dimension(0) == c.get_dimension(1));
+    ONEDAL_ASSERT(c.has_mutable_data());
+
+    const auto nd = c.get_dimension(0);
+    const auto kd = a.get_dimension(0);
+    const auto* const a_ptr = a.get_data();
+    auto* const c_ptr = c.get_mutable_data();
+    const auto a_str = a.get_leading_stride();
+    const auto c_str = c.get_leading_stride();
```

Comment:
This name looks like a string for me

Reply:
1) As far as I know there are almost no strings in oneDAL library. Thus it seems strange to confuse stride with string in such context.
2) Unfortunately there are many places where this wording is already in use.